Task 2 te 347 provide web fixt

kbayraktar opened the pull request:
fixture Methods added

Commit:
Fixtures expanded for testing angular JS-App.

Commit:
proxy username and password system properties added.

kpanier commented on web/src/main/java/org/testeditor/fixture/web/WebDriverFixture.aml:
```diff
@@ -58,6 +58,23 @@ interaction type click {
 	method = WebDriverFixture.clickOn(element)	
 }
+interaction type moveToElementAndClick {
+	label = "moveToElementAndClick"
```
The label is equals to the type name, so we can drop it here

kbayraktar replied:
this can be changed immediately.

kpanier commented on web/src/main/java/org/testeditor/fixture/web/WebDriverFixture.java:
```diff
@@ -41,6 +46,7 @@
 	private WebDriver driver;
 	private Logger logger = LogManager.getLogger(WebDriverFixture.class);
 	private String exeuteScript = null;
+	private List<WebElement> allSelectedOptions;
```
Why have we this variable? This gives the fixture another state.

kbayraktar replied:
this can be changed immediately.

kpanier commented on web/src/main/java/org/testeditor/fixture/web/WebDriverFixture.java:
```diff
@@ -194,6 +216,45 @@ public String readValue(String elementLocator) {
+		Thread.sleep(300);
+		Map<String, String> namesOfAllSelectedOptions = new HashMap<String, String>();
+		Select selection = new Select(getWebElement(elementLocator));
+		allSelectedOptions = selection.getAllSelectedOptions();
```
I don't see any reason to have allSelectedOptions as an object memeber. Method scope shuld be enough.

kbayraktar replied:
this can be changed immediately.

kpanier commented on web/src/main/java/org/testeditor/fixture/web/WebDriverFixture.java:
```diff
@@ -194,6 +216,45 @@ public String readValue(String elementLocator) {
+		WebElement element = getWebElement(elementLoacator);
+		boolean enabled = element.isEnabled();
+		if (!enabled) {
+			return null; 
```
Why this one? We return true or null?

kbayraktar replied:
This is a workaround until Gunther changes the Test-Editor to handle this.

kpanier replied:
The boolean check is allready in the development branch

Commit:
[TE-347] changes implemented see following entries:
- WebDriverFixture.java - checkEnabled Method returns now Boolean values and not null
- WebDriverFixture.aml label changed
